pkg/sysregistries/system_registries.go

```diff
@@ -29,8 +31,17 @@ type tomlConfig struct {
 
 // Reads the global registry file from the filesystem. Returns
 // a byte array
-func readRegistryConf() ([]byte, error) {
-	configBytes, err := ioutil.ReadFile(systemRegistriesConfPath)
+func readRegistryConf(ctx *types.SystemContext) ([]byte, error) {
+	dirPath := systemRegistriesConfPath
+	if ctx != nil {
+		if ctx.RegistriesDirPath != "" {
+			dirPath = ctx.SystemRegistriesConfPath
+		}
+		if ctx.RootForImplicitAbsolutePaths != "" {
```

Comment:
`else if` here (the template, `registriesDirPath`, has a `return` for each of the cases); `RegistriesDirPath` has the highest priority.